model/project_ref.go

```diff
@@ -2244,55 +2266,57 @@ func (p *ProjectRef) GetGithubProjectConflicts() (GithubProjectConflicts, error)
-		return false, nil
-	}
-	catcher := grip.NewBasicCatcher()
-	allEnabledProjects, err := GetEnabledProjects()
-	if err != nil {
-		return true, errors.Wrap(err, "getting number of projects")
-	}
-	allEnabledProjectsId := []string{}
-	for _, p := range allEnabledProjects {
-		allEnabledProjectsId = append(allEnabledProjectsId, p.Id)
-	}
-	// No need to validate if project was enabled to begin with.
-	if utility.StringSliceContains(allEnabledProjectsId, projectId) {
-		return false, nil
+		return http.StatusOK, nil
 	}
 
-	// Only error if we are enabling a previously disabled project.
-	shouldError := false
-	pRef, err := findOneProjectRefQ(byId(projectId))
-	if err != nil {
-		return true, errors.Wrapf(err, "getting project '%s", projectId)
-	}
-	if pRef != nil {
-		// Defaulting to repo
-		if projectRef == nil {
-			projectRef, err = GetProjectRefMergedWithRepo(*pRef)
-			if err != nil {
-				return true, errors.Wrapf(err, "getting merged project for '%s'", projectId)
-			}
+	isNewProject := originalMergedRef == nil
+	catcher := grip.NewBasicCatcher()
+	if shouldValidateTotalProjectLimit(isNewProject, originalMergedRef) {
+		allEnabledProjects, err := GetNumberOfEnabledProjects()
+		if err != nil {
+			return http.StatusInternalServerError, errors.Wrap(err, "getting number of projects")
+		}
+		if allEnabledProjects >= config.ProjectCreation.TotalProjectLimit {
+			catcher.Errorf("total enabled project limit of %d reached", config.ProjectCreation.TotalProjectLimit)
 		}
-		shouldError = !pRef.IsEnabled() && projectRef.IsEnabled()
 	}
 
-	if len(allEnabledProjects) >= config.ProjectCreation.TotalProjectLimit {
-		catcher.Errorf("total project limit of %d reached", config.ProjectCreation.TotalProjectLimit)
-	}
-	if !config.ProjectCreation.IsExceptionToRepoLimit(projectRef.Owner, projectRef.Repo) {
-		enabledOwnerRepoProjects, err := GetEnabledProjectsForOwnerRepo(projectRef.Owner, projectRef.Repo)
+	if shouldValidateOwnerRepoLimit(isNewProject, config, originalMergedRef, mergedRefToValidate) {
+		enabledOwnerRepoProjects, err := GetNumberOfEnabledProjectsForOwnerRepo(mergedRefToValidate.Owner, mergedRefToValidate.Repo)
 		if err != nil {
-			return true, errors.Wrapf(err, "getting number of projects for '%s/%s'", projectRef.Owner, projectRef.Repo)
+			return http.StatusInternalServerError, errors.Wrapf(err, "getting number of projects for '%s/%s'", mergedRefToValidate.Owner, mergedRefToValidate.Repo)
 		}
-		if len(enabledOwnerRepoProjects) >= config.ProjectCreation.RepoProjectLimit {
-			catcher.Errorf("owner repo limit of %d reached for '%s/%s'", config.ProjectCreation.RepoProjectLimit, projectRef.Owner, projectRef.Repo)
+		if enabledOwnerRepoProjects >= config.ProjectCreation.RepoProjectLimit {
+			catcher.Errorf("owner repo limit of %d reached for '%s/%s'", config.ProjectCreation.RepoProjectLimit, mergedRefToValidate.Owner, mergedRefToValidate.Repo)
```

Comment:
since this error will go to users, should be a bit clearer: "enabled project limit" might be clearer than "owner repo limit" (we specify owner/repo at the end so that clarifies it i think)

Reply:
done